datafusion/physical-optimizer/src/enforce_distribution.rs

```diff
@@ -1036,9 +1036,9 @@ pub fn replace_order_preserving_variants(
     if is_sort_preserving_merge(&context.plan) {
         let child_plan = Arc::clone(&context.children[0].plan);
         // It's safe to unwrap because `CoalescePartitionsExec` supports `fetch`.
-        context.plan = CoalescePartitionsExec::new(child_plan)
-            .with_fetch(context.plan.fetch())
-            .unwrap();
+        context.plan = Arc::new(
```

Comment:
Could you please rebase main and fix more places using `unwrap`? There's some such code added yesterday